actionpack/test/dispatch/prefix_generation_test.rb

```diff
@@ -327,6 +346,14 @@ def setup
       assert_equal "http://www.example.com/awesome/blog/posts/1", path
     end
 
+    test "[OBJECT] generating engine's route with polymorphic_url from singular resource" do
+      path = engine_object.polymorphic_path(Author.new)
+      assert_equal "/awesome/blog/author", path
+
+      path = engine_object.polymorphic_url(Author.new, host: "www.example.com")
+      assert_equal "http://www.example.com/awesome/blog/author", path
+    end
```

Comment:
Do we also need to test the routes for persisted authors?

Reply:
I would test that the `resolve` keeps working.

Reply:
I think if we make this change we should probably just remove the reference to `resolve` in the Docs since it won't be necessary anymore.

Reply:
I don't think we need to test that author is persisted as that is not part of this test.

Reply:
Adding it to its own test.